DB-10878 more server-side logging.

hatyo opened the pull request:
In this PR we add more DRDA logging on the server. The purpose of this is to log as many exceptions as possible for easier postmortems.

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -97,6 +97,8 @@
 
     private static final Logger AUDITLOG =Logger.getLogger("splice-audit");
 
+    private static final Logger log = Logger.getLogger(DRDAConnThread.class);
```
_log_ -> _LOG_ for consistency

hatyo replied:
done.

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -306,13 +309,22 @@ private void parsePRDDTA() throws DRDAProtocolException
+    }
+
+    private void debug(String message) {
+        if(log.isDebugEnabled()) {
```
The reason we redundantly check if Debug/Trace levels are enabled outside the logging methods is to avoid argument evaluation which is done unconditionally and may have quite noticeable overhead.
Introducing an intermediate convenience method apparently defeats the purpose. 
The existing _trace()_ method doesn't conform to that principle and should be reworked (separately from this PR; also to use Logger). It should not be considered as acceptable still.

hatyo replied:
Yes, I also think we should change `trace()` to use our `Logger` but I would do it separately since it is orthogonal to this change.

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -380,6 +393,8 @@ public void run() {
                 // TODO: Could make use of Throwable.addSuppressed here when
                 //       compiled as Java 7 (or newer).
                 try {
+                    debug("encountered error " + error.getMessage());
```
Use `LOG.debug(message, error)`. It will also print the `error.getMessage()` and the stack trace.

hatyo replied:
done.

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -1188,6 +1200,9 @@ private void processCommands() throws DRDAProtocolException
                     SanityManager.THROWASSERT("Unexpected exception after " +
                             "codePoint "+cpStr, sqle);
+                } catch (Exception e) {
+                    server.consoleExceptionPrint(e);
```
Use `LOG`. Ideally we'll have just one logging system and it won't be the Derby one.

hatyo replied:
Yes, that's definitely something we should do, I didn't want to introduce that here because I thought it is out of scope. Will create a JIRA for this at least.

For now, I would prefer to keep it like this, because the error will appear anyway (although in a different log file).

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -8697,14 +8714,24 @@ private void closeSession()
+        catch (Exception e)
+        {
+            // log the exception and rethrow.
+            debug("encountered exception while closing session "+ getSessionName());
```
`LOG.debug(message, error)`

hatyo replied:
done.

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java:
```diff
@@ -8725,6 +8752,7 @@ private void handleException(Exception e)
             if (e instanceof DRDAProtocolException) {
                 // protocol error - write error message
                 sendProtocolException((DRDAProtocolException) e);
+                server.consoleExceptionPrintTrace(e);
```
`LOG.debug(message, error)`

hatyo replied:
I would make it part of this [JIRA](https://splicemachine.atlassian.net/browse/DB-10893).

OlegMazurov commented on db-drda/src/main/java/com/splicemachine/db/impl/drda/NetworkServerControlImpl.java:
```diff
@@ -3973,6 +3972,10 @@ void addSession(Socket clientSocket) throws Exception {
                     thread = new DRDAConnThread(session, this, getTimeSlice(),
                                                 getLogConnections());
+                    thread.setUncaughtExceptionHandler((t, e) -> {
+                        consoleMessage("tracing unhandled exception originating from thread: '" + t.getName() + "'", true);
```
`LOG.debug(message, error)`

hatyo replied:
I would make it part of this [JIRA](https://splicemachine.atlassian.net/browse/DB-10893).

jpanko1 reviewed (commented):
I approve contingent on Oleg's comments being resolved.

hatyo commented:
@OlegMazurov thanks for the feedback, I addressed most of your comments, I would address the rest of the comments related to using LOG instead of Derby's own logging mechanisms such as `trace`, `consoleMessage`, and `consoleExceptionPrintTrace` in a separate [JIRA](https://splicemachine.atlassian.net/browse/DB-10893), since it is orthogonal to the purpose of the this ticket.

carolp-503 reviewed (approved):
+1
